docs_src/design/ucr/0005-Record-and-Replay.md

```diff
@@ -0,0 +1,48 @@
+## Record and Replay
+### Submitters
+- Lenny Goodell (Intel)
+- Jim White (IOTech)
+
+## Change Log
+- [pending](URL of PR) (2022-08-11)
+
+### Market Segments
+- Any/All
+
+### Motivation
+Currently one must have physical devices and appropriate environment to produce real device data (Event/Readings) into an EdgeX solution for other EdgeX services (Core Data, App Services, eKuiper Rules Engine) to consume. This is often not the case when someone is developing/testing one of these consuming EdgeX services. A good example of this is the **[RFID LLRP Inventory App Service](https://github.com/edgexfoundry/app-rfid-llrp-inventory)** . In order to test this service it is dependent on the [RFID LLRP Device Service](https://github.com/edgexfoundry/device-rfid-llrp-go), physical LLRP RFID readers, RFID Tags and environment where these are deployed. Having a way to record EdgeX the Event/Readings from an actual deployment that then could be replayed in development environment for testing would be very valuable. 
+
+Other potential uses are:
+
+- “Replayable data” could be used to validate a new or updated EdgeX service behaves like the existing one (EdgeX certification). Or the data could be offered by component providers to give users an example of the type of data generated from one of their device services.
+- EdgeX Ready applicants could use a record feature to capture data from their device service and the committee could explore that in relation to their profile to make sure they seemed to know what they were doing.
+- Record and replay can be used to reproduce a reported bug.
+- Record and replay could serve as a means to backup data in the event of data loss.
```

Comment:
This might be a stretch - like having a periodic backup? Otherwise, how could you rely on it?
Then you have to worry about "how far back".

Reply:
I think this is a stretch, but the capability would allow a service on the other end of it to make the determination about how far back and where to put the data.  As a requirement, its a "lesser" priority, but interesting to keep it is as a stretch.

Reply:
Note that I didn't include this in the requirements. This is in the motivation section and is indirectly enabled by the export requirement, which is a poor mans backup. ;-)